src/mono/wasm/runtime/startup.ts

```diff
@@ -470,6 +466,8 @@ async function instantiate_wasm_module(
 
         replace_linker_placeholders(imports);
         const assetToLoad = await loaderHelpers.wasmDownloadPromise.promise;
+
+        await ensureUsedWasmFeatures();
```

Comment:
we could actually bomb even before downloading the wasm file, few lines earlier.

Reply:
I have fired it before awaiting wasm download, but awaited after. Makes sense?